queue.cpp

```diff
@@ -0,0 +1,99 @@
+#include <iostream>
+#include <stack>
+#include <algorithm>
+#include <functional>
+
+using namespace std;
+
+template<typename T, typename F = function<T(T, T)>>
+class AggQueue {
+private:
+    stack<pair<T, T>> inputStack, outputStack;
```

Comment:
Coupling of actual and dynamic values is not a good thing tbh, as i described to other student:
E.g. you want `emplace(Args&&)` functuon, you cant do this one with coupled stacks.